applications/MeshingApplication/custom_processes/metrics_spr_error_process.cpp

```diff
@@ -0,0 +1,593 @@
+
+        noalias(b) -= mPenaltyNormal*prod(trans(p_k),trans(N_k))*itPatchNode->GetValue(CONTACT_PRESSURE);
+    }
+
+    // Neighboring nodes:
+    for( auto& i_neighbour_node : itPatchNode->GetValue(NEIGHBOUR_NODES)) {
+        if (i_neighbour_node.Has(CONTACT_PRESSURE)){
+            const array_1d<double, 3>& normal_neigh_node = i_neighbour_node.GetValue(NORMAL);
+            const double x_patch = itPatchNode->X();
+            const double y_patch = itPatchNode->Y();
+            const double x_neigh = i_neighbour_node.X();
+            const double y_neigh = i_neighbour_node.Y();
+            p_k(0,1)= x_neigh-x_patch;
+            p_k(0,2)= y_neigh-y_patch;
+            p_k(1,4)= x_neigh-x_patch;
+            p_k(1,5)= y_neigh-y_patch;
+            p_k(2,7)= x_neigh-x_patch;
+            p_k(2,8)= y_neigh-y_patch;
+            N_k(0,0) = normal_neigh_node[0]*normal_neigh_node[0];
+            N_k(0,1) = normal_neigh_node[1]*normal_neigh_node[1];
+            N_k(0,2) = 2*normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,0) = normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,1) = -normal_neigh_node[0]*normal_neigh_node[1];
+            T_k1(0,2) = normal_neigh_node[1]*normal_neigh_node[1]-normal_neigh_node[0]*normal_neigh_node[0];
+
+            noalias(A1) = prod(trans(p_k),trans(N_k));
+            noalias(A2) = prod(N_k,p_k);
+            noalias(A) += mPenaltyNormal*prod(A1, A2);
+
+            noalias(A1) = prod(trans(p_k),trans(T_k1));
+            noalias(A2) = prod(T_k1,p_k);
+            noalias(A) += mPenaltyTangent*prod(A1, A2);
+
+            noalias(b) += mPenaltyNormal*prod(trans(p_k),trans(N_k))*i_neighbour_node.GetValue(CONTACT_PRESSURE);
+        }
+    }
+
+    // Computing coefficients a: A*a=b
+    KRATOS_INFO_IF("SPRMetricProcess", mEchoLevel > 3) << A << std::endl;
+    
+    Vector coeff(SigmaSize * (TDim+1));
+    Vector b_vector = row(b, 0);
+    mpLinearSolver->Solve(A, coeff, b_vector);
+
+    for (IndexType j = 0; j < SigmaSize;++j){
+        p_k(j,j*(TDim + 1) + 1) = itNode->X() - itPatchNode->X();
+        p_k(j,j*(TDim + 1) + 2) = itNode->Y() - itPatchNode->Y();
+        if (TDim == 3)
+            p_k(j,j*(TDim + 1) + 3) = itNode->Z() - itPatchNode->Z();
+    }
+    
+    BoundedMatrix<double, SigmaSize*(TDim + 1), 1> coeff_matrix;
+    for (IndexType i=0; i<SigmaSize*(TDim + 1); ++i)
+        coeff_matrix(i, 0) = coeff[i];
+    
+    noalias(sigma) = prod(p_k, coeff_matrix);
+
+    rSigmaRecovered = row(sigma,0);
+    
+    KRATOS_INFO_IF("SPRMetricProcess", mEchoLevel > 1) <<" Recovered pressure: "<< prod(N_k,sigma) <<", LM: "<<itNode->GetValue(CONTACT_PRESSURE)<<std::endl;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void SPRMetricProcess<TDim>::ComputeElementSize(ElementItType itElement)
```

Comment:
why are tet and tri treated in a special way?
=> comment is missing

Reply:
These are the unique cases threated anyway, the other case is for the future in case more types of geometries were supported by the remesher library

Reply:
Can you please add this explanation as comment?